### PR TITLE
Rebuild embed.js to use React and handle embeddable Rays (e.g. using Beam Checkout)

### PR DESCRIPTION
**Pull request by robbieaverill**
Summary of changes:

* Rebuild in React with Tailwind for CSS
* Modal opens when an anchor or button has `data-chec-embed-url="..."` attribute
* CSS is lazy loaded once JS is loaded
* Frame closes when you press ESC (unless focus is within the iframe)

image

Todo:

* Add linting
* Put build/deployment into CI
* Handle multiple instances on a page gracefully

Example embed code to test this with:

```html
<script type="text/javascript" src="https://cdn.chec.io/v3/embed.js" async></script>
<a href="#" data-chec-embed-url="https://staging-likable-kale.commercerays.com">Buy now!</a>
<div id="chec-embed"></div>
```

### CODE REVIEW

File: src/Embed.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      if (e.key === 'Escape') {
+        setVisible(false);
+      }
+    };
+
+    window.addEventListener('keyup', closeSpy);
+    return () => {
+      window.removeEventListener('keyup', closeSpy);
+    };
+  }, [visible]);
```
Comment (ScopeyNZ) [inline src/Embed.tsx:87]: This effect will register a new event handler for `keyup` every time visible is `true`, but doesn't clean them up. The cleanup function runs when the component is unmounted, not when the effect is run again (AFAIK).

File: src/Embed.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  if (!visible || !url) {
+    return null;
+  }
+
+  return (
+    <div className="chec-embed">
+      <div className="chec-embed__container">
+        <iframe src={url} width="100%" height="100%" title="Buy now" className="chec-embed__frame">
+          Your browser does not support frames, please visit {url} directly.
```
Comment (ScopeyNZ) [inline src/Embed.tsx:97]: Do you want to make `{url}` actually a link here?

**Review by ScopeyNZ — COMMENTED** [review]
What about supporting a `data-ray-id` or something?

Looks good code-wise. I'll play around with it a bit in the morning.

File: src/Embed.tsx (hunk 2 above)
Comment (robbieaverill) [reply src/Embed.tsx:97]: I think it's very unlikely that browsers won't support iframes, so no

File: src/Embed.tsx (hunk 1 above)
Comment (robbieaverill) [reply src/Embed.tsx:87]: I tried this and it complained about `visible` not being provided as a dependency, so leaving it the way it is

**Comment by robbieaverill** [comment]
Approved verbally by @ScopeyNZ

**Pull request merged**